httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyWebSocket.java

```diff
@@ -115,6 +115,7 @@ public void onWebSocketClose(int statusCode, String reason) {
 
   @Override
   public void onWebSocketConnect(Session session) {
+    this.webSocketSession = session;
```

Comment:
Could you please elaborate on this change?
I have a fade memory having trouble doing sth similar (but can't remember exactly right now, maybe accessing the session before it was set)

Reply:
When I started looking at my failed jetty test, I reviewed the implementation more. This change mirrors the logic Jetty's WebSocketAdapter - in particular making the session volatile.

Reply:
Also onWebSocketConnect is called before the websocketclient.connect future is completed.

Reply:
Yes, as I said I can't remember exactly what or why I placed that there.
Maybe it was just an issue with the future chaining method, dunno, anyway doesn't matter.